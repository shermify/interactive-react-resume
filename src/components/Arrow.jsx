import React from 'react';
import { Link } from 'react-scroll';

const Arrow = () => (
  <Link to="profile" className="arrow-link" smooth offset={-50} duration={500}>
    <svg className="arrows">
      <path className="a1" d="M0 0 L50 12 L100 0" />
      <path className="a2" d="M0 15 L50 27 L100 15" />
      <path className="a3" d="M0 30 L50 42 L100 30" />
    </svg>
  </Link>
);
export default Arrow;
