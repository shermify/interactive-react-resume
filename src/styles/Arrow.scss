.arrow-link {
  bottom: 0;
  cursor: pointer;
  left: 0;
  position: absolute;
  text-align: center;
  width: 100%;

  .arrows {
    padding-top: 10px;
    height: 58px;
    width: 100px;

    path {
      stroke: purple;
      fill: transparent;
      stroke-width: 6px;
      animation: arrow 2s infinite;
      -webkit-animation: arrow 2s infinite;
    }
    @keyframes arrow {
      0% {
        opacity: 0;
      }

      40% {
        opacity: 1;
      }

      80% {
        opacity: 0;
      }

      100% {
        opacity: 0;
      }
    }

    path.a1 {
      animation-delay: -1s;
      -webkit-animation-delay: -1s;
    }

    path.a2 {
      animation-delay: -0.5s;
      -webkit-animation-delay: -0.5s;
    }

    path.a3 {
      animation-delay: 0s;
      -webkit-animation-delay: 0s;
    }

  }

}
