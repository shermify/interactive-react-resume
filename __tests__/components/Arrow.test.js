import React from 'react';
import renderer from 'react-test-renderer';
import Arrow from '../../src/components/Arrow';

describe('<Arrow />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Arrow />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
