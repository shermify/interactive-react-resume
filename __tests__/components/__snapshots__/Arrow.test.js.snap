// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Arrow /> renders correctly 1`] = `
<a
  className="arrow-link"
  onClick={[Function]}
>
  <svg
    className="arrows"
  >
    <path
      className="a1"
      d="M0 0 L50 12 L100 0"
    />
    <path
      className="a2"
      d="M0 15 L50 27 L100 15"
    />
    <path
      className="a3"
      d="M0 30 L50 42 L100 30"
    />
  </svg>
</a>
`;
